test/clj/chronograph/domain/timer_test.clj

```diff
@@ -0,0 +1,480 @@
+      (let [unstarted-timer-by-admin (timer/create! db/datasource
+                                                    admin-id
+                                                    task-id
+                                                    "A note by an admin.")
+            unstarted-timer-by-member (timer/create! db/datasource
+                                                     member-id
+                                                     task-id
+                                                     "A note by a member.")]
+        (is (nil? (timer/update-note! db/datasource
+                                      member-id
+                                      (:timers/id unstarted-timer-by-admin)
+                                      "A sneaky note by a member."))
+            "A member cannot overwrite an admin's note.")
+        (is (nil? (timer/update-note! db/datasource
+                                      admin-id
+                                      (:timers/id unstarted-timer-by-member)
+                                      "A sneaky note by an admin."))
+            "An admin cannot overwrite a member's note.")))))
+
+
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Start Timer Tests
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+
+(deftest start-timer-test
+  (testing "starting a timer"
+    (let [member-id (:member-id @test-context)
+          timer (timer/create! db/datasource
+                               member-id
+                               (:task-id-1 @test-context)
+                               "A sample note.")]
+      (is (s/valid? :time-spans/time-span
+                    (timer/start! db/datasource
+                                  member-id
+                                  (:timers/id timer)))
+          "returns a valid TimeSpan. This TimeSpan represents the currently-running state of the timer.")
+      (is (nil? (timer/start! db/datasource
+                              member-id
+                              (:timers/id timer)))
+          "does nothing. An already-started timer remains started."))))
+
+(deftest start-timer-isolation-by-user-test
+  (let [admin-id (:owner-id @test-context)
+        member-id (:member-id @test-context)
+        task-id (:task-id-1 @test-context)]
+    (testing "Users trying to start each others' timers."
+      (let [unstarted-timer-by-admin (timer/create! db/datasource
+                                                    admin-id
+                                                    task-id
+                                                    "A note by an admin.")
+            unstarted-timer-by-member (timer/create! db/datasource
+                                                     member-id
+                                                     task-id
+                                                     "A note by a member.")]
+        ;; first ensure un-started timers remain un-started
+        (is (nil? (timer/start! db/datasource
```

Comment:
We can ignore roles here since our code doesn't use them anywhere here yet.

Reply:
Fixed; ignoring roles.